Adamant/Services/DataProviders/CoreDataRealationMapper.swift

```diff
@@ -19,34 +19,40 @@ final class CoreDataRealationMapper: CoreDataRealationMapperProtocol {
         self.stack = stack
     }
 
-    func mapReactionRelationship(transaction: RichMessageTransaction) async -> [String] {
+    func mapReactionRelationship(transaction: RichMessageTransaction) async -> String? {
         let privateContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
         privateContext.parent = stack.container.viewContext
-
+        
         return await privateContext.perform {
             guard let id = transaction.getRichValue(for: RichContentKeys.react.reactto_id) else {
-                return []
+                return nil
             }
-
-            let processedIds: [String] = [id]
-
+            
             let chatRequest = NSFetchRequest<ChatTransaction>(entityName: "ChatTransaction")
             chatRequest.predicate = NSPredicate(format: "transactionId == %@", id)
             chatRequest.fetchLimit = 1
-
+            
             do {
                 if let chatTrs = try privateContext.fetch(chatRequest).first {
```

Comment:
Nit: 

since try is irrelevant for the outcome, consider using try? , removing do { } catch {} and just returning id